packages/chopsticks/src/blockchain/index.ts

```diff
@@ -90,14 +91,18 @@ export class Blockchain {
     this.#maxMemoryBlockCount = maxMemoryBlockCount
   }
 
-  #registerBlock(block: Block) {
+  async #registerBlock(block: Block, saveToDB = true) {
     // if exceed max memory block count, delete the oldest block
     if (this.#blocksByNumber.size === this.#maxMemoryBlockCount) {
       const firstKey = this.#blocksByNumber.keys().next().value
       this.#blocksByNumber.delete(firstKey)
     }
     this.#blocksByNumber.set(block.number, block)
     this.#blocksByHash[block.hash] = block
+    // save to db
+    if (saveToDB) {
```

Comment:
can you remove this from here and call it only when needed?

Reply:
any specific reason? I actually want to save block when register, otherwise this logic would be sacttered around

Reply:
not every block should go to DB, only those we create

Reply:
setHead for example should save block